tests/local/test_text_parsing.py

```diff
@@ -0,0 +1,175 @@
+                                           Michael Keith Shenk, for himself.
+                                           Shari Salu, for respondent.
+                                         GUSTAFSON, Judge: The Internal Revenue Service (IRS)
+                                      determined a deficiency of $3,136 in the 2009 Federal income
+                                      tax of petitioner Michael Keith Shenk. Mr. Shenk petitioned
+                                      this Court, pursuant to section 6213(a), 1 for redetermination
+                                      of the deficiency. After Mr. Shenk’s concession that he
+                                      received but did not report $254 in dividend income, the
+                                      issue for decision is whether Mr. Shenk is entitled to a
+                                      dependency exemption deduction for one of his children
+                                      under section 151(c), a child tax credit for that child under
+                                      section 24(a), and head-of-household filing status under sec-
+                                      tion 2(b)(1). On these issues, we hold for the IRS.
+                                                                          FINDINGS OF FACT
+
+                                      The judgment of divorce
+                                        Mr. Shenk was married to Julie Phillips, and they have
+                                      three minor children—M.S., W.S., and L.S. They divorced in
+                                      2003. The family court’s ‘‘Judgment of Absolute Divorce’’ pro-
+                                         1 Unless otherwise indicated, all citations of sections refer to the Internal
+
+                                      Revenue Code (26 U.S.C.) in effect for the tax year at issue, and all cita-
+                                      tions of Rules refer to the Tax Court Rules of Practice and Procedure.
+
+
+
+
+VerDate Nov 24 2008   10:59 Jul 11, 2014   Jkt 372897   PO 20012   Frm 00002   Fmt 3857   Sfmt 3857   V:\FILES\BOUND VOL. WITHOUT CROP MARKS\B.V.140\SHENK   JAMIE
+                                      202                 140 UNITED STATES TAX COURT REPORTS                                   (200)
+
+
+                                      vided: that Ms. Phillips was ‘‘awarded primary residential
+                                      custody’’ of the parties’ three children; and that Mr. Shenk
+                                      would be liable for child support payments; but that, as to
+                                      dependency exemptions—""",
+                "140 T.C. 200",
+            ),
+        )
+        site = tax.Site()
+        for q, a in test_pairs:
+
+            results = site.extract_from_text(q)
+            self.assertEqual(results["normalized_citation"], a, msg="Success")
+            print "✓", results["normalized_citation"]
```

Comment:
Nit: extra line return here. 

`msg` is what you want it to say when it fails. You can leave it blank and it does pretty good things, or if you want to populate it, you should give it something *better* than the default. But saying "success" in `msg` is wrong.

Reply:
Nixed and nixed